### ISSUE TITLE
NLU training triggers only on specific keyword(s) but does not generalize

### ISSUE THREAD

**Issue by ewagner70** [RasaHQ/rasa_core#403]
**Rasa Core version**: 0.9.0a2
**Rasa NLU version**: 0.12.2
**Python version**: 3.6
**Tensorflow version**: tensorflow-gpu 1.7.0
**Operating system** (windows, osx, ...): Ubuntu 17.10
**Issue**: See example below:

George: hi
Bot: hey there None!
George: my name is eric
Bot: Good to know you, None ## here should come "eric" instead of "None"
George: i am called eric
Bot: Good to know you, None  ## here should come "eric" instead of "None"
George: my parents call me eric
Bot: Good to know you, eric  ## here now comes finally  "eric" instead of "None", due to obviously well trained trigger keyword "call"

**Content of domain file** (if used & relevant): see attached zip file
Thx in advance! KR Eric
[test_rasa.zip](https://github.com/RasaHQ/rasa_core/files/1939693/test_rasa.zip)

**Comment by akelad** [comment]
sorry, just saw that it was included in the zip file. For entity extraction to work properly, you need a lot more examples than you have. There's a tool that can help you generate a few more training examples https://github.com/rodrigopivi/Chatito. But the recommendation of `ner_spacy` still stands

**Comment by akelad** [comment]
Your pipeline seems fine, yes
1) Yes, you can use both, they're complementary. `ner_spacy` uses a pretrained model of commonly known entities, and `ner_crf` trains entities from your training data. 
2) What do you mean with the sequence? The order of the pipeline isn't really important, if that's what you mean.
3) As for general recommendations:  this depends on what you want to achieve. with entity recognizers, if you have entities that aren't available in pretrained models, then you need `ner_crf`. If you need to recognise stuff like time/emails/duration etc then you can use `ner_duckling`. If you want to recognise stuff like locations, you can use `ner_spacy`. The latter two don't require you to label entities in your training data. 
When it comes to intent classifiers, the spacy one you're using is fine for your case. If it gets a bit more complicated, or a lot of out of vocabulary words are present, then you can consider the `tensorflow_embedding` pipeline. There's more information about all of this in the NLU docs

**Comment by ewagner70** [comment]
Thank you for your explanation, Akela!
interesting, that the sequence of that pipeline doesn't matter. Am I right assuming that all of these pipeline items run de-facto separately and produce their respective results on the same input?

**Comment by ewagner70** [comment]
(3) important further question: in order to avoid ner_cerf to train person names, you suggested to use ner_spacy --> How can I use ner_spacy to automatically fill in entity/slot "username" defined in my domain.yml?  Or has the entity/slot name be "PERSON" to be used by ner_spacy?

**Comment by akelad** [comment]
Have you looked at this page? https://nlu.rasa.com/pipeline.html

**Comment by ewagner70** [comment]
Yes, of course ... what I meant, are real examples for each of the possible pipelines - I'm currently playing around with the results and do a lot of trial and error ... any good/best/worst case examples would be really helpful and save a lot of time ... if rasa cannot provide that, you can close that threat.
KR Eric

**Comment by akelad** [comment]
Sure, I'd say `ner_spacy` and `ner_duckling` are self explanatory, they're relevant when you have common entities in your data like `time`, `location` etc. There's a link to the duckling docs in our docs, and you can read more about the `ner_spacy` entities here: https://spacy.io/usage/linguistic-features#entity-types
As for tensorflow, alan wrote a blog post here: https://medium.com/rasa-blog/supervised-word-vectors-from-scratch-in-rasa-nlu-6daf794efcd8
